packages/library/src/commonMain/kotlin/io/realm/RealmConfiguration.kt

```diff
@@ -173,10 +206,26 @@ public class RealmConfiguration private constructor(
          * installed by default that will redirect to the common logging framework on the platform, i.e.
          * LogCat on Android and NSLog on iOS.
          */
-        fun log(level: LogLevel = LogLevel.WARN, customLoggers: List<RealmLogger> = emptyList()) = apply {
-            this.logLevel = level
-            this.userLoggers = customLoggers
-        }
+        fun log(level: LogLevel = LogLevel.WARN, customLoggers: List<RealmLogger> = emptyList()) =
+            apply {
+                this.logLevel = level
+                this.userLoggers = customLoggers
+            }
+
+        /**
+         * Setting this will change the behavior of how migration exceptions are handled. Instead of throwing an
+         * exception the on-disc Realm will be cleared and recreated with the new Realm schema.
+         *
+         * **WARNING!** This will result in loss of data.
+         */
+        fun deleteRealmIfMigrationNeeded() = apply { this.deleteRealmIfMigrationNeeded = true }
+
+        /**
+         * Sets the schema version of the Realm. This must be equal to or higher than the schema version of the existing
+         * Realm file, if any. If the schema version is higher than the already existing Realm, a migration is needed.
```

Comment:
Maybe add the rules of migrations here. E.g. we only support adding/removing classes. Anything more requires the `deleteRealmIfMigrationNeeded` flag (and then maybe a link to the github issue).